odoo/osv/expression.py

```diff
@@ -868,27 +868,27 @@ def child_of_domain(left, ids, left_model, parent=None, prefix=''):
                 (when available), or as an expanded [(left,in,child_ids)] """
             if not ids:
                 return [FALSE_LEAF]
-            left_model = left_model.with_context(active_test=False)
-            if left_model._parent_store:
+            left_model_sudo = left_model.sudo().with_context(active_test=False)
+            if left_model_sudo._parent_store:
                 domain = OR([
                     [('parent_path', '=like', rec.parent_path + '%')]
-                    for rec in left_model.sudo().browse(ids)
+                    for rec in left_model_sudo.browse(ids)
                 ])
             else:
                 # recursively retrieve all children nodes with sudo(); the
                 # filtering of forbidden records is done by the rest of the
                 # domain
-                parent_name = parent or left_model._parent_name
-                if (left_model._name != left_model._fields[parent_name].comodel_name):
-                    raise ValueError(f"Invalid parent field: {left_model._fields[parent_name]}")
+                parent_name = parent or left_model_sudo._parent_name
+                if (left_model_sudo._name != left_model_sudo._fields[parent_name].comodel_name):
```

Comment:
Doesn't really make sense to use `sudo` for `_name`, `_parent_name`, `_fields` or `_parent_store`, this is just generating noise in the diff

Reply:
True. I updated that.